docs/src/pages/guides/composition/composition.md

```diff
@@ -81,23 +81,26 @@ Let's change our `ListItemLink` to the following:
 ```jsx
 import { Link as RouterLink } from 'react-router-dom';
 
-class ListItemLink extends React.Component {
-  renderLink = React.forwardRef((itemProps, ref) => (
-    // with react-router-dom@^5.0.0 use `ref` instead of `innerRef`
-    <RouterLink to={this.props.to} {...itemProps} innerRef={ref} />
-  ));
-
-  render() {
-    const { icon, primary, secondary, to } = this.props;
-    return (
-      <li>
-        <ListItem button component={this.renderLink}>
-          {icon && <ListItemIcon>{icon}</ListItemIcon>}
-          <ListItemText inset primary={primary} secondary={secondary} />
-        </ListItem>
-      </li>
-    );
-  }
+function ListItemLink(props) {
+  const { icon, primary, to } = props;
+
+  const renderLink = React.useMemo(
```

Comment:
I think I might revert these changes, since I'm not sure if the doc explanations entirely match the changes being made here.

> renderLink will now always reference the same component.

This would only be true if the `to` prop hasn't changed between renders. If it has, then it's probably going to return a different component.

Reply:
Looks good to me.